aioasuswrt/connection.py

```diff
@@ -132,37 +137,40 @@ async def async_run_command(self, command, first_try=True):
     async def async_connect(self):
         """Connect to the ASUS-WRT Telnet server."""
         async with self._io_lock:
-            self._reader, self._writer = await asyncio.open_connection(
-                self._host, self._port
+            await self._async_connect()
+
+    async def _async_connect(self):
+        self._reader, self._writer = await asyncio.open_connection(
+            self._host, self._port
+        )
+
+        # Process the login
+        # Enter the Username
+        try:
+            await asyncio.wait_for(self._reader.readuntil(b"login: "), 9)
+        except asyncio.IncompleteReadError:
+            _LOGGER.error(
+                "Unable to read from router on %s:%s" % (self._host, self._port)
             )
+            return
+        except TimeoutError:
+            _LOGGER.error("Host timeout.")
```

Comment:
Probably best to also add a disconnect here, similar as for the async_run_command.

Reply:
Done